VA: Add integration test for HTTP timeouts.

cpu opened the pull request:
Also update `TestHTTPTimeout` to test with the `SimplifiedVAHTTP`
feature flag enabled.

jsha commented on va/va_test.go:
```diff
@@ -378,33 +379,58 @@ func TestHTTPTimeout(t *testing.T) {
+				err := features.Set(map[string]bool{"SimplifiedVAHTTP": true})
+				test.AssertNotError(t, err, "Failed to set SimplifiedVAHTTP feature flag")
+				defer features.Reset()
+				// Simplified VA HTTP error messages don't include the port # when it is
```
I think there's a copy-paste error here (`#`)

cpu replied:
I meant "port #" as in "port number" but its probably clearer to just say "number".

cpu replied:
Fixed in c9dc957

cpu commented:
Looks like this branch breaks the `test_revoke_by_privkey` integration test somehow - I will investigate.

cpu commented:
> Looks like this branch breaks the `test_revoke_by_privkey` integration test somehow - I will investigate.

I changed the `pebble-challtestsrv` from binding to `:5002` for HTTP to `10.88.88.88:5002` but left the `FAKE_DNS` set to `127.0.0.1`. Fixed in b3d944c.

jsha commented:
(test-only)
